packages/react-components/react-toast/src/state/vanilla/toast.ts

```diff
@@ -1,13 +1,58 @@
+import { ValidatedToastOptions } from '../types';
+
 // TODO convert to closure
 export class Toast {
-  public running = false;
-  public onUpdate: () => void = () => null;
+  public running: boolean;
+  public onUpdate: () => void;
+  private targetDocument: Document;
+  private toastElement?: HTMLElement;
+  private pauseOnWindowBlur: ValidatedToastOptions['pauseOnWindowBlur'];
+  private pauseOnHover: ValidatedToastOptions['pauseOnHover'];
 
-  public play() {
-    this.running = true;
+  constructor(targetDocument: Document, options: ValidatedToastOptions) {
+    this.running = false;
+    this.onUpdate = () => null;
+    this.targetDocument = targetDocument;
+
+    const { pauseOnHover, pauseOnWindowBlur } = options;
+    this.pauseOnHover = pauseOnHover;
+    this.pauseOnWindowBlur = pauseOnWindowBlur;
+
+    if (this.pauseOnWindowBlur) {
+      this.targetDocument.defaultView?.addEventListener('focus', this.play);
+      this.targetDocument.defaultView?.addEventListener('blur', this.pause);
+    }
```

Comment:
Is it a good idea to add a listener to every toast?

Reply:
it's kind of necessary in the current design since the play state is owned exclusively by the toast itself and not the toaster. Each toast can also configure these independently of other toasts

Reply:
I don't think there would be enough toasts on a page to worry about this now, but we can certain change this if it becomes a problem. I don't see this becoming a part of the public API ever